crates/uv/src/commands/build.rs

```diff
@@ -198,11 +227,103 @@ async fn build_impl(
             .root()
             .clone();
 
-        Source::Directory(Cow::Owned(project))
+        vec![Source::Directory(Cow::Owned(project))]
+    } else if all {
+        if matches!(src, Source::File(_)) {
+            return Err(anyhow::anyhow!(
+                "Cannot specify a `--all` when building from a file"
+            ));
+        }
+        let workspace = match workspace {
+            Ok(ref workspace) => workspace,
+            Err(err) => {
+                return Err(
+                    anyhow::anyhow!("`--all` was provided, but no workspace was found")
+                        .context(err),
+                )
+            }
+        };
+        workspace
+            .packages()
+            .values()
+            .map(|package| Source::Directory(Cow::Owned(package.root().clone())))
```

Comment:
I think we need to omit virtual packages, and probably fail if there are no members to build.

Reply:
Added this.